include/devicetree/pm_state.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2020 Intel corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DEVICETREE_PM_STATE_H_
+#define ZEPHYR_INCLUDE_DEVICETREE_PM_STATE_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <power/power_state.h>
+
+/**
+ * @defgroup devicetree-pm_states Power Management states device tree macros
+ * @{
+ */
+
+/**
+ * @brief Construct a pm_state_info from 'pm_states' property at index 'i'
+ *
+ * @param node_id A node identifier compatible with pm-states
+ * @param i index of pm_states prop which type is 'phandles'
+ * @return pm_state_info item from 'pm_states' property at index 'i'
+ */
+#define DT_PM_STATE_INFO_ITEM_BY_IDX(node_id, i)                    \
+	{                                                           \
+		.state = DT_ENUM_IDX(DT_PHANDLE_BY_IDX(node_id,     \
+					pm_states, i), pm_state),    \
+		.min_residency_us = DT_PROP_BY_PHANDLE_IDX_OR(node_id, \
+				pm_states, i, min_residency_us, -1),   \
+	},
+
+/**
+ * @brief Length of 'pm-states' property which type is 'phandles'
+ *
+ * @param node_id A node identifier compatible with pm-states
+ * @return length of 'pm-states' property which type is 'phandles'
+ */
+#define DT_PM_STATE_ITEMS_LEN(node_id) DT_PROP_LEN(node_id, pm_states)
+
+/**
+ * @brief Macro function to construct enum pm_state item in UTIL_LISTIFY
+ * extension.
+ *
+ * @param child child index in UTIL_LISTIFY extension.
+ * @param node_id A node identifier compatible with pm-states
+ * @return macro function to construct a pm_state_info
+ */
+#define DT_PM_STATE_INFO_ITEMS_FUC(child, node_id) \
```

Comment:
What the heck does `_FUC` mean?  I see this in the nuvoton npcx macros, but it makes no sense there either.

Reply:
I guess _FUNC would have been more adequate here.

Reply:
Or maybe `_ULFUNC` as it's supposed to be a function for `UTIL_LISTIFY`?

Either would be an improvement.

Reply:
^^ @mbolivar @galak do you want a standard naming convention DT macros that are intended to be used in `UTIL_LISTIFY`?